Illegal Type Check, updated default illegal types, added memberModifiers...

alexkravin opened the pull request:
... option, issue #567 

According to #567 

Updated default illegal types list and UTs, added information about them to docs.

Added new option <b>memberModifiers</b> which allows to only verify methods and fields with specified access modifiers.
It is string array as I think it's more obvious for user to specify option as "protected, public" than "TokenTypes.LITERAL_PROTECTED, TokenTypes.LITERAL_PUBLIC"

Added UT and input for testing sources with new option.

romani commented:
thanks that you try to analyse problem and try to make life of user easier, but

1) 
> isVerifiableMethodOrVariable

I do not like name as where is not reference to variable or method inside if method. please name it exactly as what it do.

2) 
What is default value for memmberModifier field ? 


3)
Example of how we do other Check in that case:
http://checkstyle.sourceforge.net/config_design.html#VisibilityModifier
http://checkstyle.sourceforge.net/config_naming.html#AbbreviationAsWordInName

Please pay attention to:
http://checkstyle.sourceforge.net/config_javadoc.html#AtclauseOrder
<property name="target" value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF,
    METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/>

so it not that complicated to use TokenTypes values in option value, but it is completely exact behavior 
and we could alow user to do logic base in "final" and "static" .... any other modifiers without Check changes.

alexkravin commented:
1) Changed to <b>isVerifiable()</b> in future it can be used for not only methods or variables.

2) <b>null</b> as it's specified in xdocs

3) Rewrote it as TokenTypes constants param, yes, it's flexible in using, because user can specify any modifier, in UT there is example of specifying <b>private, protected, static</b> modifiers.
